### PR TITLE
ST-261 - GA tagging for tax code change edge case of missing payroll numbers

### PR DESCRIPTION
**Pull request by amaalali**
- [x] Unit tests passing
- [x] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala [hunk 1]
```diff
@@ -23,14 +23,29 @@ import uk.gov.hmrc.tai.model.domain.{TaxCodeChange, TaxCodeRecord}
...
+case class TaxCodeChangeViewModel(pairs: TaxCodePairs,
+                                  changeDate: LocalDate,
+                                  scottishTaxRateBands: Map[String, BigDecimal],
+                                  gaDimensions: Option[Map[String, String]])
 
 object TaxCodeChangeViewModel extends TaxCodeDescriptor {
-  def apply(taxCodeChange: TaxCodeChange, scottishTaxRateBands: Map[String, BigDecimal])(implicit messages: Messages): TaxCodeChangeViewModel = {
+
+  def apply(taxCodeChange: TaxCodeChange, scottishTaxRateBands: Map[String, BigDecimal]): TaxCodeChangeViewModel = {
+    def moreThanTwoSecondaryWithoutPayroll(records: Seq[TaxCodeRecord]): Boolean = records.count(record => !record.primary && record.payrollNumber.isEmpty) >= 2
```
Comment (SeanArmstrong) [inline app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala]: Is this really what we want.

Is not the moreThanTwoSecondaryWithoutPayrollWithSameEmployerName?

File: app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala [hunk 2]
```diff
@@ -23,14 +23,29 @@ import uk.gov.hmrc.tai.model.domain.{TaxCodeChange, TaxCodeRecord}
...
+
+
+
     val taxCodePairs = TaxCodePairs(taxCodeChange.previous, taxCodeChange.current)
     val changeDate = taxCodeChange.mostRecentTaxCodeChangeDate
 
-    TaxCodeChangeViewModel(taxCodePairs, changeDate, scottishTaxRateBands)
+    val gaDimension = if (moreThanTwoSecondaryWithoutPayroll(taxCodeChange.current) || moreThanTwoSecondaryWithoutPayroll(taxCodeChange.previous)) {
+      Some(Map("taxCodeChangeEdgeCase" -> "Yes"))
```
Comment (SeanArmstrong) [inline app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala]: Any scope to make this true/false

**Comment by SeanArmstrong** [comment]
https://github.com/hmrc/govuk-template/pull/48

Is this the best way of doing things? Seems like there should be a hook rather than govuk knowing about specific dimensions

File: app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala (hunk 1 above)
Comment (amaalali) [reply app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala]: So same name also? True will update the requirements on the tickets

Comment (SeanArmstrong) [reply app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala]: may be worth checking that this is ok

File: app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala [hunk 3]
```diff
@@ -50,4 +55,20 @@ object TaxCodeChangeViewModel extends TaxCodeDescriptor {
...
+    def moreThanTwoSecondaryWithoutPayroll(records: Seq[TaxCodeRecord]): Boolean = {
+       records.groupBy(_.employerName).values.exists(taxCodeRecords =>
+         taxCodeRecords.count(record => !record.primary && record.payrollNumber.isEmpty) >= 2
+       )
+    }
+
+    val taxCodeChangeDateGaDimension = Map[String, String]("taxCodeChangeDate" -> taxCodeChangeDate.toString(TaiConstants.EYU_DATE_FORMAT))
+
+    if (moreThanTwoSecondaryWithoutPayroll(taxCodeChange.current) || moreThanTwoSecondaryWithoutPayroll(taxCodeChange.previous)) {
+      taxCodeChangeDateGaDimension + ("taxCodeChangeEdgeCase" -> "Yes")
```
Comment (SeanArmstrong) [inline app/uk/gov/hmrc/tai/viewModels/taxCodeChange/TaxCodeChangeViewModel.scala]: Should any of these strings be put in constants
